13_import.sql

```diff
@@ -0,0 +1,575 @@
+    RAISE NOTICE 'Impossible to assign inlets - manual edit needed.';
+  ELSE
+    IF new_lets = 0 AND old_lets > 0 THEN
+      -- request for delete because no new lets but old lets
+      RAISE NOTICE 'No new inlets but old ones - manual delete needed.';
+    ELSIF new_lets > 0 AND old_lets = 0 THEN
+      -- request for create because no old lets but new lets
+      RAISE NOTICE 'No old inlets but new ones - manual create needed.';
+    ELSE
+      IF new_lets = 1 AND old_lets = 1 THEN
+        -- update material and dimension on reach
+        UPDATE qgep_od.reach
+        SET material = NEW.inlet_3_material,
+        clear_height = NEW.inlet_3_clear_hight
+        WHERE obj_id = ( SELECT re.obj_id
+          FROM qgep_od.reach re
+          LEFT JOIN qgep_od.reach_point rp ON rp.obj_id = re.fk_reach_point_to
+          LEFT JOIN qgep_od.wastewater_networkelement wn ON wn.obj_id = rp.fk_wastewater_networkelement
+          LEFT JOIN qgep_od.vw_qgep_wastewater_structure ws ON ws.obj_id = wn.fk_wastewater_structure
+          WHERE ws.obj_id = NEW.obj_id );
+
+        -- update depth_m on reach_point
+        UPDATE qgep_od.reach_point
+        SET level = NEW.co_level - NEW.inlet_3_depth_m
+        WHERE obj_id = ( SELECT rp.obj_id
+          FROM qgep_od.reach re
+          LEFT JOIN qgep_od.reach_point rp ON rp.obj_id = re.fk_reach_point_to
+          LEFT JOIN qgep_od.wastewater_networkelement wn ON wn.obj_id = rp.fk_wastewater_networkelement
+          LEFT JOIN qgep_od.vw_qgep_wastewater_structure ws ON ws.obj_id = wn.fk_wastewater_structure
+          WHERE ws.obj_id = NEW.obj_id );
+
+        RAISE NOTICE 'Inlets updated';
+      ELSE
+        -- do nothing
+        RAISE NOTICE 'No inlets - nothing to do';
+      END IF;     
+
+      -- set inlet okay
+      UPDATE qgep_import.manhole_quarantine
+      SET inlet_okay = true
+      WHERE quarantine_serial = NEW.quarantine_serial;
+    END IF;
+  END IF;
+  RETURN NEW;
+
+  -- catch
+  EXCEPTION WHEN OTHERS THEN
+    RAISE NOTICE 'EXCEPTION: %', SQLERRM;
+    RETURN NEW;
+END; $BODY$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS after_update_try_inlet_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_update_try_inlet_update
+  AFTER UPDATE
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  WHEN ( ( NEW.inlet_okay IS NOT TRUE )
+  AND NOT( OLD.outlet_okay IS NOT TRUE AND NEW.outlet_okay IS TRUE )
+  AND NOT( OLD.structure_okay IS NOT TRUE AND NEW.structure_okay IS TRUE ) )
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_inlet_update();
+
+DROP TRIGGER IF EXISTS after_insert_try_inlet_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_insert_try_inlet_update
+  AFTER INSERT
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_inlet_update();
+
+
+CREATE OR REPLACE FUNCTION qgep_import.manhole_quarantine_try_outlet_update() RETURNS trigger AS $BODY$
```

Comment:
since the two methods are very similar I would be in favor of having the two trigger functions calling a single function with the inlet/outlet parameter.

Reply:
Thought about that as well. I'm not sure what is nicer to look at. I like it in general as well not to have code "double". But on the other hand there would be a lot of `if-else` when we combine it. I will see what I can do.

Reply:
I thought the redundancy was big enough, but you are better judge

Reply:
I decided to follow your advice anyway and make one function. The reason is that I belief it makes it easier to beware the consistency, in case we update logical functionalities in future like delete etc. of existing outlets.